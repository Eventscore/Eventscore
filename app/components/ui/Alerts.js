import React, { PropTypes } from 'react';
import {
  View,
  StyleSheet,
} from 'react-native';

// Components
import { Spacer, Text } from '@ui/';

const styles = StyleSheet.create({
  alerts: {
    left: 0,
    right: 0,
  },

  // Success
  msg: {
    right: 0,
    left: 0,
    paddingVertical: 10,
    paddingHorizontal: 10,
    borderLeftWidth: 3,
    borderColor: '#1C854C',
    backgroundColor: '#59DC9A',
  },
  msg_text: {
    textAlign: 'center',
    color: '#16693c',
    fontWeight: '500',
  },

  // Error
  msgError: {
    borderColor: '#C02827',
    backgroundColor: '#FB6567',
  },
  msgError_text: {
    color: '#7f1a1a',
  },

  // Status
  msgStatus: {
    borderColor: '#408491',
    backgroundColor: '#8EDBE5',
  },
  msgStatus_text: {
    color: '#2f606a',
  },
});

const Alerts = ({ status, success, error }) => (
  <View style={styles.alerts}>
    {!!success &&
      <View>
        <View style={[styles.msg]}>
          <Text style={[styles.msg_text]}>{success}</Text>
        </View>
        <Spacer size={20} />
      </View>
    }

    {!!status &&
      <View>
        <View style={[styles.msg, styles.msgStatus]}>
          <Text style={[styles.msg_text, styles.msgStatus_text]}>
            {status}
          </Text>
        </View>
        <Spacer size={20} />
      </View>
    }

    {!!error &&
      <View>
        <View style={[styles.msg, styles.msgError]}>
          <Text
            style={[
              styles.msg_text,
              styles.msgError_text,
            ]}
          >
            {error}
          </Text>
        </View>
        <Spacer size={20} />
      </View>
    }
  </View>
);

Alerts.propTypes = {
  status: PropTypes.string,
  success: PropTypes.string,
  error: PropTypes.string,
};

Alerts.defaultProps = {
  status: '',
  success: '',
  error: '',
};

Alerts.componentName = 'Alerts';

export default Alerts;
