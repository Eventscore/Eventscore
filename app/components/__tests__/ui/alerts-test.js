/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Alerts from '@ui/Alerts';

it('Alerts (empty) renders correctly', () => {
  const tree = renderer.create(
    <Alerts />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Sucess) renders correctly', () => {
  const tree = renderer.create(
    <Alerts success={'Hello Success'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Error) renders correctly', () => {
  const tree = renderer.create(
    <Alerts error={'Error hey'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Alerts (Status) renders correctly', () => {
  const tree = renderer.create(
    <Alerts status={'Something\'s happening...'} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
