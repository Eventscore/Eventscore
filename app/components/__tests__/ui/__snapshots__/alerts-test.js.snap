// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts (Error) renders correctly 1`] = `
<View
  style={
    Object {
      "left": 0,
      "right": 0,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#59DC9A",
            "borderColor": "#1C854C",
            "borderLeftWidth": 3,
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "right": 0,
          },
          Object {
            "backgroundColor": "#FB6567",
            "borderColor": "#C02827",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        onPress={null}
        p={false}
        style={
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "HelveticaNeue",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 18,
            },
            Array [
              Object {
                "color": "#16693c",
                "fontWeight": "500",
                "textAlign": "center",
              },
              Object {
                "color": "#7f1a1a",
              },
            ],
          ]
        }
      >
        Error hey
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 1,
          "left": 0,
          "marginTop": 19,
          "right": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Alerts (Status) renders correctly 1`] = `
<View
  style={
    Object {
      "left": 0,
      "right": 0,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#59DC9A",
            "borderColor": "#1C854C",
            "borderLeftWidth": 3,
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "right": 0,
          },
          Object {
            "backgroundColor": "#8EDBE5",
            "borderColor": "#408491",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        onPress={null}
        p={false}
        style={
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "HelveticaNeue",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 18,
            },
            Array [
              Object {
                "color": "#16693c",
                "fontWeight": "500",
                "textAlign": "center",
              },
              Object {
                "color": "#2f606a",
              },
            ],
          ]
        }
      >
        Something's happening...
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 1,
          "left": 0,
          "marginTop": 19,
          "right": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Alerts (Sucess) renders correctly 1`] = `
<View
  style={
    Object {
      "left": 0,
      "right": 0,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#59DC9A",
            "borderColor": "#1C854C",
            "borderLeftWidth": 3,
            "left": 0,
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "right": 0,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        onPress={null}
        p={false}
        style={
          Array [
            Object {
              "color": "#222222",
              "fontFamily": "HelveticaNeue",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 18,
            },
            Array [
              Object {
                "color": "#16693c",
                "fontWeight": "500",
                "textAlign": "center",
              },
            ],
          ]
        }
      >
        Hello Success
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 1,
          "left": 0,
          "marginTop": 19,
          "right": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Alerts (empty) renders correctly 1`] = `
<View
  style={
    Object {
      "left": 0,
      "right": 0,
    }
  }
/>
`;
